libraries/AC_WPNav/AC_Circle.cpp

```diff
@@ -222,17 +205,18 @@ bool AC_Circle::update()
 
     } else {
         // set target position to center
-        Vector3f target;
-        target.x = _center.x;
-        target.y = _center.y;
-        target.z = target_z_cm;
-
-        // update position controller target
-        _pos_control.set_pos_target(target);
+        pos.x = _center.x;
+        pos.y = _center.y;
 
         // heading is same as _angle but converted to centi-degrees
         _yaw = _angle * DEGX100;
     }
+    pos.z = target_z_cm;
```

Comment:
We could change this line to, "pos.z = _center.z + terr_offset" and then remove this section a bit higher up

```
    // calculate z-axis target
    float target_z_cm;
    if (_terrain_alt) {
        target_z_cm = _center.z + terr_offset;
    } else {
        target_z_cm = _center.z;
    }
```

Reply:
done